jdbc/src/java/org/apache/hive/jdbc/HiveConnection.java

```diff
@@ -1329,21 +1328,32 @@ private String getSessionValue(String varName, String varDefault) {
     return varValue;
   }
 
-  // use socketTimeout from jdbc connection url. Thrift timeout needs to be in millis
-  private void setupLoginTimeout() {
-    String socketTimeoutStr = sessConfMap.getOrDefault(JdbcConnectionParams.SOCKET_TIMEOUT, "0");
-    long timeOut = 0;
-    try {
-      timeOut = Long.parseLong(socketTimeoutStr);
-    } catch (NumberFormatException e) {
-      LOG.info("Failed to parse socketTimeout of value " + socketTimeoutStr);
-    }
-    if (timeOut > Integer.MAX_VALUE) {
-      loginTimeout = Integer.MAX_VALUE;
-    } else if (timeOut < 0) {
-      loginTimeout = 0;
+  // For loginTimeout, prefer to use loginTimeout (in milliseconds) from jdbc connection url, and
+  // fallback to use DriverManger#getLoginTimeout (in seconds).
+  // For socketTimeout, use socketTimeout (in milliseconds) from jdbc connection url if present.
+  private void setupTimeout() {
+    long _loginTimeout = 0;
```

Comment:
nit: we prefer not to use `_` in the variable names

Reply:
Any suggestions here? What about use `loginTimeout` and use `this.loginTimeout` to ref member variable?

Reply:
That's the hardest part of the development. Coming up with good variable names 😄 
`loginTimeoutFromConf` or `confLoginTimeout` or `parsedLoginTimeout` or `longLoginTimeout`?

But if you have better ideas then I would happy to see them 😄